### PR TITLE
Final cq manny

### PR DESCRIPTION
**Pull request by MannyB6**
This will incorporate the CQ part of the game.

### CODE REVIEW

**Comment by MannyB6** [comment]
This was a group project that all 4 of us worked on to make CQ work.

File: src/main/java/cs361/battleships/models/Ship.java [hunk 1]
```diff
@@ -65,32 +109,79 @@ public String getKind() {
...
+
+
+
 		return result;
 	}
 
 	@JsonIgnore
 	public boolean isSunk() {
-		return getOccupiedSquares().stream().allMatch(s -> s.isHit());
+		return (getOccupiedSquares().stream().allMatch(s -> s.isHit())) || (getOccupiedSquares().stream().anyMatch(s -> s.isHit() && s.isCQ()));
```
Comment (PembleC) [inline src/main/java/cs361/battleships/models/Ship.java]: Nice addition!

File: src/main/java/cs361/battleships/models/Ship.java [hunk 2]
```diff
@@ -65,32 +109,79 @@ public String getKind() {
...
+//if(sunkCounter >= 3){
+//			return true;
+//		}
+//		else
+//			return false;
+	}
+
+	@JsonIgnore
+	public boolean isSunk2() {
+		if(getOccupiedSquares().stream().anyMatch(s -> s.isHitCQ())){
```
Comment (PembleC) [inline src/main/java/cs361/battleships/models/Ship.java]: Never used...

Comment (MannyB6) [reply src/main/java/cs361/battleships/models/Ship.java]: I guess I could've done a check with isSunk() and isSunk2()...but it works!

**Comment by PembleC** [comment]
Group Work!

I think we need to do some cleaning to remove your old code, but it looks good and passes the tests! Way to go TEAM 11

**Pull request merged**